internal/resources/cluster/backupschedule/resource_backup_schedule.go

```diff
@@ -0,0 +1,144 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package backupschedule
+
+import (
+	"context"
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/authctx"
+	clienterrors "github.com/vmware/terraform-provider-tanzu-mission-control/internal/client/errors"
+	backupschedulemodels2 "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/cluster/backupschedule"
+	"strings"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceBackupSchedule() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceBackupScheduleCreate,
+		ReadContext:   resourceBackupScheduleRead,
+		UpdateContext: resourceBackupScheduleUpdate,
+		DeleteContext: resourceBackupScheduleDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceBackupScheduleImporter,
+		},
+		Schema: backupScheduleResourceSchema,
+	}
+}
+
+func resourceBackupScheduleCreate(ctx context.Context, data *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+	model := tfModelResourceConverter.ConvertTFSchemaToAPIModel(data, []string{})
+
+	request := &backupschedulemodels2.VmwareTanzuManageV1alpha1ClusterDataprotectionScheduleRequest{
+		Schedule: model,
+	}
+
+	_, err := config.TMCConnection.BackupScheduleService.BackupScheduleResourceServiceCreate(request)
+
+	if err != nil {
+		return diag.FromErr(errors.Wrapf(err, "unable to create Tanzu Mission Control backup schedule. Name: %s, Cluster: %s",
+			request.Schedule.FullName.Name, request.Schedule.FullName.ClusterName))
+	}
+
+	diags = resourceBackupScheduleRead(ctx, data, m)
+
+	return diags
+}
+
+func resourceBackupScheduleRead(ctx context.Context, data *schema.ResourceData, m interface{}) (diags diag.Diagnostics) {
+	config := m.(authctx.TanzuContext)
+	backupScheduleFn := tfModelResourceConverter.ConvertTFSchemaToAPIModel(data, []string{nameKey, clusterNameKey}).FullName
+	getResponse, err := config.TMCConnection.BackupScheduleService.BackupScheduleResourceServiceGet(backupScheduleFn)
+
+	if err != nil || getResponse.Schedule == nil {
+		return diag.Errorf("Couldn't read backup schedule. Name: %s, Cluster: %s", backupScheduleFn.Name, backupScheduleFn.ClusterName)
+	} else {
+		err = tfModelResourceConverter.FillTFSchema(getResponse.Schedule, data)
+
+		if err != nil {
+			return diag.Errorf("Couldn't read backup schedule. Name: %s, Cluster: %s", backupScheduleFn.Name, backupScheduleFn.ClusterName)
+		}
+
+		data.SetId(fmt.Sprintf("%s/%s", backupScheduleFn.ClusterName, backupScheduleFn.Name))
```

Comment:
as mentioned in the data protection [PR](https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/270#pullrequestreview-1643567481) I don't think just cluster name is sufficient.

Reply:
Let's decide what would be a valid UID for a cluster and I will change both to follow the same convention.
Pay attention that in this resource it is also concatenated with the Event name so if you have any suggestions to that also, it will be great

Reply:
As commented in enable data protection, I decide to go with managment_cluster_name/provisioner_name/cluster_name/backup_schedule_name to keep it consistent with import command